transport/src/main/java/io/netty/channel/AbstractChannelHandlerContext.java

```diff
@@ -940,11 +958,24 @@ public ChannelPromise voidPromise() {
     }
 
     final void setRemoved() {
-        handlerState = REMOVED;
+        handlerState = REMOVE_COMPLETE;
     }
 
     final void setAdded() {
-        handlerState = ADDED;
+        for (;;) {
+            int oldState = handlerState;
+            // Ensure we never update when the handlerState is REMOVE_COMPLETE already.
+            // oldState is usually ADD_PENDING but can also be REMOVE_COMPLETE when an EventExecutor is used that is not
+            // exposing ordering guarantees.
+            if (oldState == REMOVE_COMPLETE || HANDLER_STATE_UPDATER.compareAndSet(this, oldState, ADD_COMPLETE)) {
+                return;
+            }
+        }
+    }
+
+    final void setBeforeAdded() {
+        boolean updated = HANDLER_STATE_UPDATER.compareAndSet(this, INIT, ADD_PENDING);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "updated". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1854)